src/conn/conn_sweep.c

```diff
@@ -139,7 +140,8 @@ __sweep_expire(WT_SESSION_IMPL *session)
 		    F_ISSET(dhandle, WT_DHANDLE_DEAD))
 			continue;
 		if (dhandle->session_inuse != 0 ||
-		    now <= dhandle->timeofdeath + conn->sweep_idle_time)
+		    now <= dhandle->timeofdeath + conn->sweep_idle_time ||
+		    conn->sweep_idle_time == 0)
```

Comment:
I'd move this test to early in ```__sweep_expire```. The function is a no-op if we aren't expiring handles.

I'd change this to be:

```
/* Never expire handles if idle timeout is disabled. */
if (conn->sweep_idle_time == 0)
    return (0);
```

Just before the call to ```__wt_seconds```.